MParT/Utilities/Serialization.h

```diff
@@ -0,0 +1,65 @@
+#ifndef MPART_SERIALIZATION_H
+#define MPART_SERIALIZATION_H
+
+#include <cereal/types/base_class.hpp>
+#include <cereal/archives/binary.hpp>
+#include <Kokkos_Core.hpp>
+
+#include "MParT/Utilities/ArrayConversions.h"
+
+namespace cereal {
+    template <typename ScalarType, typename MemorySpace, typename Archive>
+    std::enable_if_t<traits::is_output_serializable<BinaryData<ScalarType>, Archive>::value, void> save(
+        Archive &ar, Kokkos::View<ScalarType*, MemorySpace> const &vec) {
```

Comment:
Should this be for arbitrary memory spaces or should we restrict it to HostSpace only?   Instead of throwing an exception in the `load` function, we could just restrict ourselves to the hostspace.   

This will also only work for views with the default layout.  We often use strided layouts, especially when interacting with other languages like python, and should allow for saving with arbitrary layouts and then loading back into either `LayoutLeft` or `LayoutRight`.   I'm not sure it makes sense to load back into a strided view.